Cleanup: Netty TLS & Config Features

isaacrivriv opened the pull request:
Work through organizing and cleaning up TODOs for Netty TLS cleanup
fixes #31359

isaacrivriv committed:
First attempt to move config to a shared bundle between http and ssl and other transport implementations

isaacrivriv committed:
Reached working config bundle with netty tls bundle using ssl options

isaacrivriv committed:
Added necessary changes for http to work with new tls api

isaacrivriv committed:
Updated JMS features and bundles to include the necessary configuration bundles for Netty TLS

isaacrivriv committed:
Updated SIP feature and bundle to include the necessary configuration for Netty TLS

isaacrivriv committed:
Updated wsoc feature and bundle to include the necessary configuration for Netty TLS

mrsaldana commented on dev/io.openliberty.netty.internal.tls.impl/src/io/openliberty/netty/internal/tls/impl/handler/LibertySslHandler.java:
```diff
@@ -0,0 +1,118 @@
+        engine().getSSLParameters().setUseCipherSuitesOrder(enforceCipherOrder);
+        this.suppressLogError = SslOption.SUPPRESS_HANDSHAKE_ERRORS.parse(sslOptions);
+        this.maxLogEntries = SslOption.SUPPRESS_HANDSHAKE_ERRORS_COUNT.parse(sslOptions);
+        System.out.println("Got options enforceCipherOrder:" + enforceCipherOrder + ", suppressHandshakeError: " + suppressLogError + ", maxCount: " + maxLogEntries );
```
Cleanup sysout

mrsaldana commented on dev/io.openliberty.netty.internal.tls.impl/src/io/openliberty/netty/internal/tls/impl/handler/package-info.java:
```diff
@@ -0,0 +1,11 @@
+/*******************************************************************************
+ * Copyright (c) 2021 IBM Corporation and others.
```
Fix age range

mrsaldana commented on dev/io.openliberty.netty.internal.tls.impl/src/io/openliberty/netty/internal/tls/impl/handler/LibertySslHandler.java:
```diff
@@ -0,0 +1,118 @@
+/*******************************************************************************
+ * Copyright (c) 2024 IBM Corporation and others.
```
Change to year range

mrsaldana commented on dev/io.openliberty.netty.internal/src/io/openliberty/netty/internal/tls/NettyTlsProvider.java:
```diff
@@ -10,13 +10,14 @@
 
 import java.util.Map;
-import io.netty.handler.ssl.SslContext;
+import io.netty.channel.Channel;
```
Adjust copyright

mrsaldana commented on dev/com.ibm.ws.sipcontainer/src/com/ibm/ws/sip/stack/transport/sip/netty/SipOutboundConnLink.java:
```diff
@@ -76,12 +76,11 @@ public SipOutboundConnLink(String peerHost, int peerPort, SipInboundChannel sipI
                   final ChannelPipeline pipeline = ch.pipeline();
                   if (isSecure) {
-                      SslContext context = GenericEndpointImpl.getTlsProvider().getOutboundSSLContext(GenericEndpointImpl.getSslOptions(), peerHost, Integer.toString(peerPort));
+                      SslHandler handler = GenericEndpointImpl.getTlsProvider().getOutboundSSLContext(GenericEndpointImpl.getSslOptions(), peerHost, Integer.toString(peerPort), ch);
```
This class needs copyright range update

mrsaldana commented on dev/com.ibm.ws.sipcontainer/src/com/ibm/ws/sip/stack/transport/netty/GenericTCPChain.java:
```diff
@@ -183,8 +180,9 @@ protected void initChannel(Channel ch) throws Exception {
             if (isTLS) {
-                SSLEngine engine = context.newEngine(ch.alloc());
-                pipeline.addFirst("ssl", new SslHandler(engine, false));
+                EndPointInfo ep = nettyBundle.getEndpointManager().getEndPoint(getEndpointName());
```
This class needs copyright range update

mrsaldana commented on dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/pipeline/HttpPipelineInitializer.java:
```diff
@@ -151,10 +151,9 @@ private void setupUnsecurePipeline(ChannelPipeline pipeline) {
 
-        SslContext context = getSslContext();
-        SSLEngine engine = context.newEngine(pipeline.channel().alloc());
+        SslHandler handler = getSslHandler(pipeline.channel());
```
Update copyright range on this class

mrsaldana commented on dev/io.openliberty.netty.internal.tls.impl/bnd.bnd:
```diff
@@ -24,6 +24,8 @@ Include-Resource: \
 
 Export-Package: io.openliberty.netty.internal.tls.impl
 
+Private-Package: io.openliberty.netty.internal.tls.impl.handler
```
Copyright range

mrsaldana commented on dev/io.openliberty.transport.config.internal/src/io/openliberty/transport/config/options/EndpointOption.java:
```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+ * Copyright (c) 2024 IBM Corporation and others.
```
Adjust range

mrsaldana commented on dev/io.openliberty.transport.config.internal/src/io/openliberty/transport/config/options/SslOption.java:
```diff
@@ -0,0 +1,67 @@
+/*******************************************************************************
+ * Copyright (c) 2024 IBM Corporation and others.
```
Adjust range

mrsaldana commented on dev/io.openliberty.transport.config.internal/src/io/openliberty/transport/config/options/utils/TransportConfigUtils.java:
```diff
@@ -0,0 +1,237 @@
+/*******************************************************************************
+ * Copyright (c) 2024 IBM Corporation and others.
```
Adjust range

isaacrivriv committed:
Initial cleanup of comments

isaacrivriv committed:
Updated references to new bundle
- Removed old classes in shared bundles
- Update references to new bundle
- Switch classes to appropriate bundles

pnicolucci commented on dev/com.ibm.ws.messaging.comms.server/src/com/ibm/ws/jfap/inbound/channel/NettyInboundChain.java:
```diff
@@ -347,14 +333,22 @@ protected void initChannel(Channel ch) throws Exception {
+                SslHandler handler = _commsServerFacade.getNettyTlsProvider().getInboundSSLContext(_currentConfig.sslOptions, host, port, ch);
+                if(handler == null) {
+                    if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.entry(this, tc, "initChannel","Error adding TLS Support");
+                    throw new NettyException("Problems creating SslHandler");
```
Can we add this to: https://github.com/OpenLiberty/open-liberty/issues/31761 ?

isaacrivriv replied:
Done, added to the issue

pnicolucci commented on dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/netty/NettyHttpChannelConfig.java:
```diff
@@ -197,20 +193,6 @@ private void parseHttpOptions(Map<String, Object> options) {
 
     }
 
-
```
I think it makes sense to update the copyright date range on this class as well.

pnicolucci commented on dev/com.ibm.ws.transport.http/src/io/openliberty/http/options/CompressionOption.java:
```diff
@@ -55,10 +55,4 @@ public Class<?> getValueType() {
     public ConfigType getConfigType() {
         return configType;
     }
-
```
Copyright date range update? Seems significant enough?

pnicolucci replied:
There are a number of classes with similar updates, if you don't agree that's fine just let me know. I figure since we're changing functionality and no longer overriding a method that to me is fairly significant.

isaacrivriv replied:
I'll make the update to the copyright

pnicolucci commented on dev/io.openliberty.netty.internal.tls.impl/src/io/openliberty/netty/internal/tls/impl/NettyTlsProviderImpl.java:
```diff
@@ -219,13 +219,14 @@ public SslContext getInboundSSLContext(Map<String, Object> sslOptions, String ho
         String[] protocols;
         try {
         	Properties props = createProps(sslOptions);
+            System.out.println("Got suppres: " + SslOption.SUPPRESS_HANDSHAKE_ERRORS.parse(sslOptions));
```
System.out should be removed, just add traces here for what you want?

pnicolucci commented on dev/io.openliberty.transport.config.internal/src/io/openliberty/transport/config/options/EndpointOption.java:
```diff
@@ -1,18 +1,20 @@
+
 /**
- * Represents a configuration option for an HTTP endpoint.
+ * Represents a configuration option for a transoport HTTP endpoint.
```
transport

isaacrivriv committed:
Further clean up changes
- Removed deleted package references from http2 client bundle
- Cleaned up messages from review
- Removed leftover print statements

isaacrivriv committed:
Added shared ssl translated messages missing for exception messages

isaacrivriv committed:
Fixed behavior change in Netty for handshake error logging

isaacrivriv committed:
Added missing build import for http configuration in websocket

isaacrivriv committed:
Code clean up, added missing copyrights, worked on missing tracing

isaacrivriv committed:
Added missing bundle import for ssl feature due to options change in Netty tls impl bundle

isaacrivriv committed:
Added reuse address property for xml file on ssl transport FAT

pnicolucci commented on dev/io.openliberty.transport.ssl_fat/publish/files/suppressHandshakeFailureLowCount.xml:
```diff
@@ -20,7 +20,7 @@
           httpPort="${bvt.prop.HTTP_default}"
           httpsPort="${bvt.prop.HTTP_default.secure}" >
-          <tcpOptions portOpenRetries="60"/>
+          <tcpOptions soReuseAddr="true" portOpenRetries="60"/>
```
Update copyright to include 2025 date range

pnicolucci commented on dev/io.openliberty.transport.ssl_fat/publish/files/suppressHandshakeFailureTrue.xml:
```diff
@@ -20,7 +20,7 @@
           httpPort="${bvt.prop.HTTP_default}"
           httpsPort="${bvt.prop.HTTP_default.secure}" >
-          <tcpOptions portOpenRetries="60"/>
+          <tcpOptions soReuseAddr="true" portOpenRetries="60"/>
```
Update copyright to include 2025 date range

pnicolucci commented on dev/io.openliberty.transport.ssl_fat/publish/files/suppressHandshakeFailureFalse.xml:
```diff
@@ -20,7 +20,7 @@
           httpPort="${bvt.prop.HTTP_default}"
           httpsPort="${bvt.prop.HTTP_default.secure}" >
-          <tcpOptions portOpenRetries="60"/>
+          <tcpOptions soReuseAddr="true" portOpenRetries="60"/>
```
Update copyright to include 2025 date range

isaacrivriv committed:
Added requested copyright updates
